### ISSUE TITLE
New assert methods.

### ISSUE THREAD

**Issue by osolomin90** [hdorgeval/playwright-fluent#18]
**Is your feature request related to a problem? Please describe.**
I found that next assert: await playwrightFluent.expectThatSelector(SELECTOR).isVisible();
Will fail if element is out of viewport. But in most cases I need to check is element is visible any were and then I can scroll to it,
also this behaviour with viewport is not mentioned in method description. 

**Describe the solution you'd like**
I would like to have two methods: isVisible() - checks if element visible in general and isVisibleInViewport() - which works as existing one.
**Describe alternatives you've considered**
isVisible() - works as it is. isDisplayed() - checks if element visible in general.

**Comment by hdorgeval** [comment]
Hi @osolomin90, thank you for your feedback.
Fully agree with you.
As a workaround I suggest:

```js
await p
  .expectThat(selector)
  .exists()
  .hover(selector)
  .expectThat(selector)
  .isVisible();
```

The `isVisible()` assertion returns true only if the element is visible in the current viewport. If the element is outside the viewport it always returns false; and as you mentioned the documentation is not explicit on that aspect.

If you agree, I suggest that, internally, the `isVisible()` assertion should automatically trigger a hover action once the element is in the DOM,  so that the selector will automatically appear in the screen thanks to a viewport scrolling.

I will also, as you suggest, add an `.isVisibleInViewport()` assertion that will trigger no scrolling and therefore will have the same behaviour as current `.isVisible()`.

Please let me know if my suggestion fits your needs.
Thanks

**Comment by osolomin90** [comment]
Yes it will work for me) Thanks you a lot for your work!

**Comment by hdorgeval** [comment]
Hi @osolomin90, I split the work between the just published v1.46.0 and the next to be published v1.47.0.

Unfortunately, time is a bottleneck resource for me at the moment, I think v1.47.0 will be ready in about 10 days.

**Comment by hdorgeval** [comment]
Hi @osolomin90, your request is now available in `v1.47.0`

Feel free to give any feedback, if something does not fit your needs.
